drivers/ethernet/eth_stm32_hal.c

```diff
@@ -742,6 +747,7 @@ static void rx_thread(void *arg1, void *unused1, void *unused2)
 					net_pkt_unref(pkt);
 				}
 			}
+#if IS_ENABLED(CONFIG_ETH_STM32_CARRIER_CHECK)
```

Comment:
put this inside the ``else if`` below, no need to make use of the preprocessor conditionals. the preprocessor conditionals  should only be used for things, where it is not possible otherwise.